File: 121.BestTimeToBuyAndSellStock/clywin123/maxProfit.py
class Solution:
    # @param {integer[]} prices
    # @return {integer}
    def maxProfit(self, prices):
        minstock =float("inf")
        answer = 0
        for i in range(len(prices)):
            if(prices[i]<minstock):
                minstock = prices[i]
            if(prices[i]-minstock>answer):
                answer = prices[i]-minstock
        return answer

File: 122.BestTimeToBuyAndSellStockTwo/wyrj/maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let benefit = 0;
    for (let i = 1; i < prices.length; i++) {
        const delta = prices[i] - prices[i - 1];
        if (delta > 0) benefit += delta;
    }
    return benefit;
};
